itest/rust/src/builtin_tests/containers/packed_array_test.rs

```diff
@@ -276,8 +276,52 @@ fn packed_array_insert() {
     assert_eq!(array.to_vec(), vec![3, 1, 2, 4]);
 }
 
+fn test_extend<F, I>(make_iter: F)
+where
+    F: Fn(i32) -> I,
+    I: Iterator<Item = i32>,
+{
+    // The logic in `extend()` is not trivial, so we test it for a wide range of sizes: powers of two, also plus and minus one.
+    // This includes zero. We go up to 2^12, which is 4096, because the internal buffer is currently 2048 bytes (512 `i32`s)
+    // and we want to be future-proof in case it's ever enlarged.
+    let lengths = (0..12i32)
+        .flat_map(|i| {
+            let b = 1 << i;
+            [b - 1, b, b + 1]
+        })
+        .collect::<Vec<_>>();
+    for &len_a in &lengths {
```

Comment:
Empty lines between long multi-line statements :slightly_smiling_face: 
```suggestion
        .collect::<Vec<_>>();
        
    for &len_a in &lengths {
```